packages/samples/headless-ssr/app/components/search-page.tsx

```diff
@@ -0,0 +1,24 @@
+'use client';
+
+import {SearchHydrationResult, hydrateInitialState} from '@/app/common/engine';
+import {useEffect, useState} from 'react';
+import ResultList from './result-list';
+
+// TODO: replace "any" with specific type
+export default function SearchPage({engineSnapshot}: {engineSnapshot: any}) {
+  const [hydrationResult, setHydrationResult] =
+    useState<SearchHydrationResult | null>(null);
+
+  useEffect(() => {
+    hydrateInitialState(engineSnapshot).then(({engine, controllers}) => {
+      setHydrationResult({engine, controllers});
+    });
+  }, [engineSnapshot]);
+
+  if (hydrationResult === null) {
+    return <div>Hydrating engine ..</div>;
+  } else {
+    const {engine, controllers} = hydrationResult;
+    return <ResultList engine={engine} controllers={controllers}></ResultList>;
```

Comment:
For the result list, since we didn't add the `defineResultList` function yet, I think it would be easier to just give it `engine.state.search.results`, which should (IMO) be part of both the hydrated engine and the snapshot.

Reply:
This demonstrates using `buildResultList()` with the hydrated engine. Isn't that desirable? 
I wonder if demonstrating only access of the hydrated engine and not the snapshot directly would make it simpler and reduce confusion for implementers?

Reply:
When I checked SSR was working.. must have got broken during refactoring. 
Added a TODO to create a e2e test to catch this. 
Modified to render results directly using snapshot if hydration result is null. 
But still getting a blank page. Removing loading page didn't help either.
Will investigate more tomorrow.